src/neo/Cryptography/Helper.cs

```diff
@@ -156,62 +154,57 @@ public static byte[] AES256Decrypt(this byte[] encryptedData, byte[] key, byte[]
             return decryptedData;
         }
 
-        public static byte[] ECEncrypt(byte[] message, ECPoint pubKey)
+        public static byte[] EcdhEncrypt(byte[] message, KeyPair local, ECPoint remote)
```

Comment:
There is no such `ecdhencrypt` and `ecdhdecrypt` term. You only need one single function to derive the ecdh shared key, the ecrypt and decrypt would be that of AES.

Reply:
```C#
        public static byte[] ECDHDeriveKey(KeyPair local, ECPoint remote)
        {
            ECDiffieHellman ecdh1 = ECDiffieHellmanCng.Create(new ECParameters
            {
                Curve = ECCurve.NamedCurves.nistP256,
                D = local.PrivateKey,
                Q = new System.Security.Cryptography.ECPoint
                {
                    X = local.PublicKey.EncodePoint(false)[1..][..32],
                    Y = local.PublicKey.EncodePoint(false)[1..][32..]
                }
            });
            ECDiffieHellman ecdh2 = ECDiffieHellmanCng.Create(new ECParameters
            {
                Curve = ECCurve.NamedCurves.nistP256,
                Q = new System.Security.Cryptography.ECPoint
                {
                    X = remote.EncodePoint(false)[1..][..32],
                    Y = remote.EncodePoint(false)[1..][32..]
                }
            });
            return ecdh1.DeriveKeyMaterial(ecdh2.PublicKey).Sha256();//z = r * P = r* k * G
        }
```

Reply:
Have changed.